src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -356,18 +356,76 @@ protected virtual void AddRecipients(TransactionBuildContext context)
             if (context.Recipients.Any(a => a.Amount == Money.Zero))
                 throw new WalletException("No amount specified.");
 
-            if (context.Recipients.Any(a => a.SubtractFeeFromAmount))
-                throw new NotImplementedException("Substracting the fee from the recipient is not supported yet.");
+            var totalRecipients = context.Recipients.Count(r => r.SubtractFeeFromAmount);
 
-            foreach (Recipient recipient in context.Recipients)
-                context.TransactionBuilder.Send(recipient.ScriptPubKey, recipient.Amount);
+            // If we have any recipients that require a fee to be subtracted from the amount, then
+            // calculate fee and evenly distribute it among all recipients. Any remaining fee should be
+            // subtracted from the first recipient.
+            if (totalRecipients > 0)
+            {
+                Money fee = this.CalculateFee(context);
+                var recipientFee = fee.Satoshi / totalRecipients;
+                var remainingFee = fee.Satoshi % totalRecipients;
+
+                for (int i = 0; i < context.Recipients.Count; i++)
+                {
+                    var recipient = context.Recipients[i];
+
+                    if (recipient.SubtractFeeFromAmount)
+                    {
+                        // First receiver pays the remainder not divisible by output count.
+                        long feeToSubtract = i == 0 ? remainingFee + recipientFee : recipientFee;
+                        long remainingAmount = recipient.Amount.Satoshi - feeToSubtract;
+                        if (remainingAmount <= 0)
+                            throw new WalletException("Fee is higher than amount to send.");
+
+                        context.TransactionBuilder.Send(recipient.ScriptPubKey, new Money(remainingAmount));
+                    }
+                    else
+                    {
+                        context.TransactionBuilder.Send(recipient.ScriptPubKey, recipient.Amount);
+                    }
+                }
+            }
+            else
+            {
+                foreach (Recipient recipient in context.Recipients)
+                {
+                    context.TransactionBuilder.Send(recipient.ScriptPubKey, recipient.Amount);
+                }
+            }
         }
 
         /// <summary>
         /// Use the <see cref="FeeRate"/> from the <see cref="walletFeePolicy"/>.
         /// </summary>
         /// <param name="context">The context associated with the current transaction being built.</param>
         protected void AddFee(TransactionBuildContext context)
+        {
+            if (context.Recipients.Any(r => r.SubtractFeeFromAmount))
```

Comment:
Where is this done in Core's code?

(It looks right, just wanna double check)

Reply:
Their code is much more complicated, but it is all inside a `while` loop: 
https://github.com/bitcoin/bitcoin/blob/a689c119076c7b8dc5b4dea4539e4cbf5adfb72f/src/wallet/wallet.cpp#L3015